camel-pdf: build failure in jvm mode with Apache Camel 3.0.1

From https://github.com/apache/camel-quarkus/pull/630#issuecomment-575749725

```
Caused by: java.io.IOException: COSStream has been closed and cannot be read. Perhaps its enclosing PDDocument has been closed?
	at org.apache.pdfbox.cos.COSStream.checkClosed(COSStream.java:83)
	at org.apache.pdfbox.cos.COSStream.createInputStream(COSStream.java:168)
	at org.apache.pdfbox.cos.COSStream.createInputStream(COSStream.java:163)
	at org.apache.pdfbox.pdmodel.PDPage.getContents(PDPage.java:157)
	at org.apache.pdfbox.pdfparser.PDFStreamParser.<init>(PDFStreamParser.java:92)
	at org.apache.pdfbox.contentstream.PDFStreamEngine.processStreamOperators(PDFStreamEngine.java:496)
	at org.apache.pdfbox.contentstream.PDFStreamEngine.processStream(PDFStreamEngine.java:480)
	at org.apache.pdfbox.contentstream.PDFStreamEngine.processPage(PDFStreamEngine.java:153)
	at org.apache.pdfbox.text.LegacyPDFStreamEngine.processPage(LegacyPDFStreamEngine.java:139)
	at org.apache.pdfbox.text.PDFTextStripper.processPage(PDFTextStripper.java:391)
	at org.apache.pdfbox.text.PDFTextStripper.processPages(PDFTextStripper.java:319)
	at org.apache.pdfbox.text.PDFTextStripper.writeText(PDFTextStripper.java:266)
	at org.apache.pdfbox.text.PDFTextStripper.getText(PDFTextStripper.java:227)
	at org.apache.camel.component.pdf.PdfProducer.doExtractText(PdfProducer.java:104)
	at org.apache.camel.component.pdf.PdfProducer.process(PdfProducer.java:66)
```

--- a/extensions/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelMainRecorder.java
+++ b/extensions/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelMainRecorder.java
@@ -43,6 +43,10 @@ public class CamelMainRecorder {
         main.setCamelContext(runtime.getValue());
         main.addMainListener(new CamelMainEventDispatcher());
 
+        // properties are loaded through MicroProfile Config so there's
+        // no need to look for sources.
+        main.setDefaultPropertyPlaceholderLocation("false");
+
         // xml rest/routes should be explicitly configured as an
         // additional dependency is required thus, disable auto
         // discovery

--- a/extensions/dozer/deployment/src/main/java/org/apache/camel/quarkus/component/dozer/deployment/DozerProcessor.java
+++ b/extensions/dozer/deployment/src/main/java/org/apache/camel/quarkus/component/dozer/deployment/DozerProcessor.java
@@ -35,6 +35,7 @@ import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.annotations.ExecutionTime;
 import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.AdditionalApplicationArchiveMarkerBuildItem;
 import io.quarkus.deployment.builditem.FeatureBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
@@ -61,7 +62,12 @@ class DozerProcessor {
         return new JaxbFileRootBuildItem("com/github/dozermapper/core/builder/model/jaxb");
     }
 
-    @BuildStep(applicationArchiveMarkers = { "com/github/dozermapper/core" })
+    @BuildStep
+    AdditionalApplicationArchiveMarkerBuildItem dozerArchiveMarker() {
+        return new AdditionalApplicationArchiveMarkerBuildItem("com/github/dozermapper/core");
+    }
+
+    @BuildStep
     void configureCamelDozer(BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
             BuildProducer<NativeImageResourceBuildItem> nativeImage,
             CamelDozerConfig camelDozerConfig) {
@@ -112,7 +118,11 @@ class DozerProcessor {
                 new ReflectiveClassBuildItem(false, false,
                         "com.sun.org.apache.xerces.internal.impl.dv.xs.SchemaDVFactoryImpl"));
 
-        reflectiveClass.produce(new ReflectiveClassBuildItem(true, false, DozerConfiguration.class));
+        reflectiveClass.produce(new ReflectiveClassBuildItem(
+                true,
+                false,
+                DozerConfiguration.class,
+                DozerBeanMapperConfiguration.class));
 
         if (camelDozerConfig.mappingFiles.isPresent()) {
             // Register for reflection any classes participating in Dozer mapping
@@ -155,6 +165,19 @@ class DozerProcessor {
         if (camelDozerConfig.typeConverterEnabled) {
             camelDozerRecorder.initializeDozerTypeConverter(camelContextBuildItem.getCamelContext());
         }
+    }
+
+    /*
+     * Make component/endpoint configurer discoverable in native image.
+     *
+     * TODO: should not be required once https://github.com/apache/camel-quarkus/pull/618
+     *       get merged
+     */
+    @BuildStep
+    NativeImageResourceBuildItem configurers() {
+        return new NativeImageResourceBuildItem(
+                "META-INF/services/org/apache/camel/configurer/dozer-component",
+                "META-INF/services/org/apache/camel/configurer/dozer-endpoint");
     }
 
     private URI mappingPathToURI(String mappingPath) {

--- a/extensions/netty/runtime/src/main/java/org/apache/camel/quarkus/component/netty/runtime/SubstituteNettyWorkerPoolBuilder.java
+++ b/extensions/netty/runtime/src/main/java/org/apache/camel/quarkus/component/netty/runtime/SubstituteNettyWorkerPoolBuilder.java
@@ -21,7 +21,6 @@ import com.oracle.svm.core.annotate.Substitute;
 import com.oracle.svm.core.annotate.TargetClass;
 import io.netty.channel.EventLoopGroup;
 import io.netty.channel.nio.NioEventLoopGroup;
-import org.apache.camel.component.netty.NettyHelper;
 import org.apache.camel.component.netty.NettyWorkerPoolBuilder;
 import org.apache.camel.util.concurrent.CamelThreadFactory;
 
@@ -39,8 +38,7 @@ final class SubstituteNettyWorkerPoolBuilder {
 
     @Substitute
     public EventLoopGroup build() {
-        int count = workerCount > 0 ? workerCount : NettyHelper.DEFAULT_IO_THREADS;
-        workerPool = new NioEventLoopGroup(count, new CamelThreadFactory(pattern, name, false));
+        workerPool = new NioEventLoopGroup(workerCount, new CamelThreadFactory(pattern, name, false));
         return workerPool;
     }
 }

--- a/integration-tests/pdf/src/main/java/org/apache/camel/quarkus/component/pdf/it/PdfResource.java
+++ b/integration-tests/pdf/src/main/java/org/apache/camel/quarkus/component/pdf/it/PdfResource.java
@@ -43,20 +43,19 @@ public class PdfResource {
     @Inject
     ProducerTemplate producerTemplate;
 
-    private PDDocument document;
+    private byte[] document;
 
     @Path("/createFromText")
     @POST
     @Consumes(MediaType.TEXT_PLAIN)
     @Produces(MediaType.APPLICATION_OCTET_STREAM)
     public Response createFromText(String message) throws Exception {
-
-        byte[] pdfBytes = producerTemplate.requestBody(
+        document = producerTemplate.requestBody(
                 "pdf:create?fontSize=6&pageSize=PAGE_SIZE_A5&font=Courier", message, byte[].class);
-        document = PDDocument.load(pdfBytes);
-        LOG.infof("The PDDocument has been created and contains %d bytes", pdfBytes.length);
 
-        return Response.created(new URI("pdf/extractText")).entity(pdfBytes).build();
+        LOG.infof("The PDDocument has been created and contains %d bytes", document.length);
+
+        return Response.created(new URI("pdf/extractText")).entity(document).build();
     }
 
     @Path("/appendText")
@@ -64,12 +63,12 @@ public class PdfResource {
     @Consumes(MediaType.TEXT_PLAIN)
     @Produces(MediaType.APPLICATION_OCTET_STREAM)
     public Response appendText(String message) throws Exception {
+        document = producerTemplate.requestBodyAndHeader("pdf:append", message,
+                PdfHeaderConstants.PDF_DOCUMENT_HEADER_NAME, PDDocument.load(document), byte[].class);
 
-        byte[] pdfBytes = producerTemplate.requestBodyAndHeader("pdf:append", message,
-                PdfHeaderConstants.PDF_DOCUMENT_HEADER_NAME, document, byte[].class);
-        LOG.infof("The PDDocument has been updated and now contains %d bytes", pdfBytes.length);
+        LOG.infof("The PDDocument has been updated and now contains %d bytes", document.length);
 
-        return Response.ok().entity(pdfBytes).build();
+        return Response.ok().entity(document).build();
     }
 
     @Path("/extractText")
@@ -77,6 +76,6 @@ public class PdfResource {
     @Produces(MediaType.TEXT_PLAIN)
     public String get() throws Exception {
         LOG.info("Extracting text from the PDDocument");
-        return producerTemplate.requestBody("pdf:extractText", document, String.class);
+        return producerTemplate.requestBody("pdf:extractText", PDDocument.load(document), String.class);
     }
 }

--- a/integration-tests/pdf/src/test/java/org/apache/camel/quarkus/component/pdf/it/PdfTest.java
+++ b/integration-tests/pdf/src/test/java/org/apache/camel/quarkus/component/pdf/it/PdfTest.java
@@ -70,5 +70,4 @@ class PdfTest {
         assertTrue(pdfText.contains("content to be included in the created pdf document"));
         assertTrue(pdfText.contains("another line that should be appended"));
     }
-
 }

--- a/pom.xml
+++ b/pom.xml
@@ -41,7 +41,7 @@
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 
         <ahc.version>2.10.4</ahc.version>
-        <camel.version>3.0.0</camel.version>
+        <camel.version>3.0.1</camel.version>
         <hapi.version>4.1.0</hapi.version>
         <quarkus.version>1.1.1.Final</quarkus.version>
         <httpmime.version>4.1.3</httpmime.version>
@@ -161,6 +161,17 @@
 
     <repositories>
         <repository>
+            <id>apache.releases</id>
+            <url>https://repository.apache.org/content/groups/public</url>
+            <name>Apache Releases Repo</name>
+            <snapshots>
+                <enabled>false</enabled>
+            </snapshots>
+            <releases>
+                <enabled>true</enabled>
+            </releases>
+        </repository>
+        <repository>
             <id>apache.snapshots</id>
             <url>https://repository.apache.org/snapshots/</url>
             <name>Apache Snapshot Repo</name>
@@ -174,6 +185,17 @@
     </repositories>
 
     <pluginRepositories>
+        <pluginRepository>
+            <id>apache.releases</id>
+            <url>https://repository.apache.org/content/groups/public</url>
+            <name>Apache Releases Repo</name>
+            <snapshots>
+                <enabled>false</enabled>
+            </snapshots>
+            <releases>
+                <enabled>true</enabled>
+            </releases>
+        </pluginRepository>
         <pluginRepository>
             <id>apache.snapshots</id>
             <url>https://repository.apache.org/snapshots/</url>

